### PR TITLE
refactor(modify)!: split Modify control in multiple interactions

### PR DESCRIPTION
**Pull request by danji90**
# How to

- Updated getFeatureAtPixel in ModifyControl
- Now includes selectFilter to filter selectable features

Testable on mapset:
- Log into https://editor.dev.mapset.ch/
- Switch to RBS or VBZ Corporate Design
- Draw a Picture and a few other features
- Switch to SBB Corporate Design
- The Picture should not be selectable and the cursor should be 'default' on hover
- Other feature should be selectable as always
- Switch back to RBS or VBZ
- The Picture should be selectable and the cursor should be 'pointer' on hover

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [ ] IE11 tested.
- [x] The title is formatted as a [conventional-commit](https://www.conventionalcommits.org/) message.
- [x] The title contains `WIP:` if it's necessary.
- [x] Labels applied. if it's a release? a hotfix?
- [ ] Tests added.

IMPORTANT: Squash commits before or on merge to prevent every small commit being written into the change log. The Pull Request title will be written as message for the new commit containing the squashed commits and there fore needs to be in conventional-commit format

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,7 +57,7 @@
     "doc": "jsdoc -p -r -c jsdoc_conf.json src -d doc README.md",
     "format": "prettier --write 'cypress/integration/*.js' 'src/**/*.js' && eslint 'src/**/*.js' --fix && stylelint 'style/**/*.css' 'src/**/*.css' 'src/**/*.scss' --fix",
     "lint": "eslint 'cypress/integration/**/*.js' 'src/**/*.js' && stylelint 'style/**/*.css' 'src/**/*.css' 'src/**/*.scss'",
-    "publish:beta": "yarn release -- --prerelease beta && yarn build && git push origin HEAD && yarn publish --tag beta",
+    "publish:beta": "yarn release -- --prerelease beta --skip.changelog && yarn build && git push origin HEAD && yarn publish --tag beta",
```
Comment (oterral) [inline package.json:60]: :+1

File: src/control/modify.js [hunk 2]
```diff
@@ -265,15 +264,20 @@ class ModifyControl extends Control {
...
+        if (this.selectFilter(feature, layer)) {
+          features.push(feature);
+        }
+      },
+      {
+        hitTolerance: this.hitTolerance,
+        layerFilter: this.layerFilter,
+      },
+    );
+    return features.length && features[0];
```
Comment (oterral) [inline src/control/modify.js:280]: Tip of the day (if it works;)):
```
 const feature = this.map.forEachFeatureAtPixel(
      pixel,
      (feature, layer) => {
        if (this.selectFilter(feature, layer)) {
            return feature;
         }
      },
      {
        hitTolerance: this.hitTolerance,
        layerFilter: this.layerFilter,
      },
    );
    return feature;
```

**Review by oterral — APPROVED** [review]
lgtm. Please test my tip then  you can merge

File: src/control/modify.js (hunk 2 above)
Comment (danji90) [reply src/control/modify.js:280]: I tested, seems to work well. But doesn't this maybe cause the wrong feature to be returned when there are multiple features at the same pixel?

**Pull request merged**